doc/mgmt/SONiC_IntfRangeFeature_HLD.md

```diff
@@ -0,0 +1,235 @@
+
+# Scope
+* Implementation details, syntax, and unit test cases regarding interface range command support via KLISH CLI.
+* Interface range specification will not be supported via REST/gNMI.
+* Breakout command will not be supported in interface range configuration mode.
+
+# Definition/Abbreviation
+NA
+
+# 1 Feature Overview
+
+Interface Range Specification feature support via Klish CLI will allow specification of range of interfaces to which the subsequent commands will be applied. Configuring a range of interfaces will help reduce the time and effort in configuring interfaces.
+
+## 1.1 Requirements
+
+### 1.1.1 Functional Requirements
+
+Provide ability to create/configure/show/delete range of interfaces via SONiC KLISH CLI.
+
+### 1.1.2 Configuration and Management Requirements
+* Support for *interface range* command to configure the range of interfaces in both native and alias mode.
+* Extend show and delete interface commands to support range of interfaces.
+
+### 1.1.3 Scalability Requirements
+NA
+### 1.1.4 Warm Boot Requirements
+NA
+## 1.2 Design Overview
+### 1.2.1 Basic Approach
+
+- KLISH XML changes for adding new CLI view for interface range command.
+- Use of VAR tag in KLISH XML to cache existing interfaces list.
+- Changes in KLISH actioner script to invoke APIs for range of interfaces.
+- Use **interface range** command to configure existing interfaces in the range.
+- Use **interface range create** command to create non-existing interfaces in the range.
+
+### 1.2.2 Container
+This feature is implemented within the Management Framework container.
+
+### 1.2.3 SAI Overview
+# 2 Functionality
+## 2.1 Target Deployment Use Cases
+## 2.2 Functional Description
+# 3 Design
+## 3.1 Overview
+* Creation and configuration of interfaces to be done as a single transaction.
+* Transaction for show and delete of interfaces will be per interface.
+* Modifying XML file to use correct PTYPE to support number, range, or comma-delimited list of numbers and ranges.
+Example: `ETHER_INTERFACE_RANGE` for physical interfaces<br>
+```
+<PARAM
+    name="phy-if-name"
+    help="Physical interface range"
+    ptype="ETHER_INTERFACE_RANGE"
+/>
+```
+## 3.2 DB Changes
+### 3.2.1 CONFIG DB
+### 3.2.2 APP DB
+### 3.2.3 STATE DB
+### 3.2.4 ASIC DB
+### 3.2.5 COUNTER DB
+
+## 3.3 Switch State Service Design
+### 3.3.1 Orchestration Agent
+### 3.3.2 Other Process
+## 3.4 SyncD
+## 3.5 SAI
+## 3.6 User Interface
+### 3.6.1 Data Models
+NA
+### 3.6.2 CLI
+#### 3.6.2.1 Interface range commands in native mode
+
+####  3.6.2.1.1 Ethernet interface range:
+##### Configure physical interfaces in given range
+Syntax: `interface range Ethernet port - port[,port - port...]`
+```
+sonic(config)# interface range ?
+Ethernet         Physical interface range
+Vlan             Vlan interface range
+PortChannel      PortChannel interface range
+create           Create interfaces in range
+
+sonic(config)# interface range Ethernet 1-4,5
```

Comment:
Do we need to have a space between the interface name and number?

interface range Ethernet1-4,5 <<will this work?

Reply:
with/without space both cases work, will include in HLD.